drafts/build_a_dao_on_fuel/draft.md

```diff
@@ -0,0 +1,1416 @@
+
+    // Retrieve the proposal from storage and ensure it has not been executed and has passed its deadline
+    let mut proposal = storage.proposals.get(proposal_id).try_read().unwrap();
+    require(!proposal.executed, ProposalError::ProposalExecuted);
+    require(
+        proposal.deadline < height().as_u64(),
+        ProposalError::ProposalStillActive,
+    );
+
+    // Calculate the acceptance percentage of the proposal based on yes and no votes
+    let acceptance_percentage = proposal.yes_votes * 100 / (proposal.yes_votes + proposal.no_votes);
+    
+    // Ensure that the acceptance percentage meets the required threshold for execution
+    require(
+        proposal.acceptance_percentage <= acceptance_percentage,
+        ProposalError::InsufficientApprovals,
+    );
+
+    // Mark the proposal as executed
+    proposal.executed = true;
+    storage.proposals.insert(proposal_id, proposal);
+
+    // Execute the proposal transaction using assembly code
+    asm(
+        call_data: proposal.proposal_transaction.call_data,
+        amount: proposal.proposal_transaction.amount,
+        asset: proposal.proposal_transaction.asset,
+        gas: proposal.proposal_transaction.gas,
+    ) {
+        call call_data amount asset gas;
+    }
+
+    // Log an event indicating that the proposal has been executed
+    log(ExecuteEvent {
+        user: msg_sender().unwrap(),
+        acceptance_percentage,
+        id: proposal_id,
+    })
+}
+
+```
+
+***Note:In sway, the [assembly](https://docs.fuel.network/docs/sway/advanced/assembly/#asm-block) keyword `asm` is used to create an assemly inline block, declaring an `asm` block is similar to declaring a function but in low-level code, in this function the block executes the proposal's transaction. It allows for more flexibility and control over the execution process, enabling interactions that may not be possible using higher-level code alone***.
+
+When a proposal is executed, the assembly block extracts the necessary information from the proposal's transaction `call_data`, `amount`, `asset`, and `gas` and executes the transaction using these parameters. This allows the contract to perform the proposed action, such as transferring assets or invoking another contract function.
+
+```rust
+    asm(
+      call_data: proposal.proposal_transaction.call_data,
+      amount: proposal.proposal_transaction.amount,
+      asset: proposal.proposal_transaction.asset,
+      gas: proposal.proposal_transaction.gas,
+    ) {
+      call call_data amount asset gas;
+    }
+```
+
+**Note: The [`call`](https://docs.fuel.network/docs/fuels-ts/contracts/methods/#call) keyword also used in the assmbly code block is  used to submit a real contract call transaction to the node. Here is a more in-depth read on the [call method](https://docs.fuel.network/docs/fuels-ts/contracts/methods/#call) and [call parameters](https://docs.fuel.network/docs/fuels-ts/contracts/call-parameters/)**
+
+### Unlocking all Votes
```

Comment:
Consistent capitalization of section headers is important. Use this resource to help you on that: https://capitalizemytitle.com/

Reply:
This was helpful, thanks Ben
I've modified all headers appropriately